lib/modules/platform/gerrit/utils.ts

```diff
@@ -0,0 +1,114 @@
+import { CONFIG_GIT_URL_UNAVAILABLE } from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import type { BranchStatus, PrState } from '../../../types';
+import * as hostRules from '../../../util/host-rules';
+import { parseUrl } from '../../../util/url';
+import { hashBody } from '../pr-body';
+import type { Pr } from '../types';
+import type {
+  GerritChange,
+  GerritChangeStatus,
+  GerritLabelTypeInfo,
+} from './types';
+import { TAG_PULL_REQUEST_BODY } from './types';
+
+export function getGerritRepoUrl(repository: string, endpoint: string): string {
+  // Find options for current host and determine Git endpoint
+  const opts = hostRules.find({
+    hostType: 'gerrit',
+    url: endpoint,
+  });
+
+  const url = parseUrl(endpoint);
+  if (!url) {
+    throw new Error(CONFIG_GIT_URL_UNAVAILABLE);
+  }
+  url.protocol = url.protocol.slice(0, -1);
+  url.username = opts.username ?? '';
+  url.password = opts.password ?? '';
+  url.pathname = `${url.pathname}a/${encodeURIComponent(repository)}`;
+  logger.debug(
+    { url: url.toString() },
+    'using URL based on configured endpoint'
+  );
+  return url.toString();
+}
+
+export function mapPrStateToGerritFilter(state?: PrState): string {
+  switch (state) {
+    case 'closed':
+      return 'status:closed';
+    case 'merged':
+      return 'status:merged';
+    case '!open':
+      return '-status:open';
+    case 'open':
+      return 'status:open';
+    case 'all':
+    default:
+      return '-is:wip';
+  }
+}
+
+export function mapGerritChangeToPr(change: GerritChange): Pr {
+  return {
+    number: change._number,
+    state: mapGerritChangeStateToPrState(change.status),
+    sourceBranch: extractSourceBranch(change) ?? change.branch,
+    targetBranch: change.branch,
+    title: change.subject,
+    reviewers:
+      change.reviewers?.REVIEWER?.map((reviewer) => reviewer.username!) ?? [],
```

Comment:
declare username as non optional inside interface if it's always there

Reply:
that's not always true, 
the REST-Service returns these details only if requested thru additional options

see [client.ts](https://github.com/Onesty-Tech-GmbH/renovate/blob/a75d5dc6212302a32bada45809cdd01c3ce4a434/lib/modules/platform/gerrit/client.ts#L14)

Reply:
But we ensure that it is requested? Otherwise use `is.string` filter , so we remove undefined or null.